src/core/config/openthread-core-default-config.h

```diff
@@ -265,7 +275,10 @@
  *
  */
 #ifndef OPENTHREAD_CONFIG_HEAP_INTERNAL_SIZE
-#if OPENTHREAD_CONFIG_COAP_SECURE_API_ENABLE
+#if OPENTHREAD_CONFIG_SRP_SERVER_ENABLE
+// Internal heap doesn't support size larger than 64K bytes.
+#define OPENTHREAD_CONFIG_HEAP_INTERNAL_SIZE (63 * 1024)
```

Comment:
This change is for openthread core, not only posix platform.

Reply:
We are basically setting the heap size to max here. Any thoughts on what the actual bounds are?

Reply:
Memory cost of a typical Service Instance Entry:
- **fullname**: around 40 bytes for `my-service._ipps._tcp.default.service.arpa`, and 255 bytes for maximum.
- **metadata**: around 40 bytes (`weight`, `priority`, `port` and pointers).
- **TXT data**: this value can vary from 1 byte to 65535 bytes. For typical real use cases, I think it should be around 30 ~ 200 bytes.
- Total size: 40 + 40 + 30 = 110 bytes.

Memory cost of a typical Host Instance Entry:
- **fullname**: around 30 bytes for `my-host.default.service.arpa`, and 255 bytes for maximum.
- **metadata**: around 30 bytes (`lease`, `key lease` and pointers).
- **address**: we use static array (`kMaxAddressesNum = 8`) for address list and it costs `8 * 16` bytes. I think for most cases, there will be a single address. I will change this value to `2` to allow possible 1.2 addresses (e.g. DUA).
- **ECDSA KEY**: 64 bytes.
- Total size: 30 + 30 + 32 + 64 156 bytes.


I remember that the duckhorn BR requires the SRP server to serve around 500 services from 250 clients (2 services for each client). This will definitely exceeds the maximum size of the internal heap. We can do some optimizations to the SRP server implementation to reduce the memory consumption for each service and host entry. For example, we can stores only the service instance name and service name but not the full name for each service entry. But I don't think the maximum size can be limited to the internal heap capability. I think for posix and other resourceful platforms, we should just use external heap for the Border Router.